File: Basics_with_Python/pythonBasicBook/!PythonBasicCourse/4_For Loop/10_For Loop - Exercise/More Exercise2/11. Odd - Even Position .py
import sys
odd_min = sys.maxsize
odd_max = - sys.maxsize
even_min = sys.maxsize
even_max = - sys.maxsize
odd_sum = 0
even_sum = 0
odd_count = 0
even_count = 0
n = int(input())
for n in range(1, n+1):
    number = float(input())
    if n % 2 != 0:
        odd_count += 1
        odd_sum += number
        if number <= odd_min:
            odd_min = number
        if number >= odd_max:
            odd_max = number
    else:
        even_count += 1
        even_sum += number
        if number <= even_min:
            even_min = number
        if number >= even_max:
            even_max = number

print(f"OddSum={odd_sum:.2f},")
if odd_count >= 1:
    print(f"OddMin={odd_min:.2f},")
    print(f"OddMax={odd_max:.2f},")
else:
    print(f"OddMin=No,")
    print(f"OddMax=No,")
print(f"EvenSum={even_sum:.2f},")
if even_count >= 1:
    print(f"EvenMin={even_min:.2f},")
    print(f"EvenMax={even_max:.2f}")
else:
    print(f"EvenMin=No,")
    print(f"EvenMax=No")

File: Basics_with_JavaScript/910_For_Loop/More_Exersises/11._Odd__Even_Position.js
function vladi(array){
    let numbers = Number(array[0]);
    let oddSum = 0; let oddMin = 0; let oddMax = 0; let evenSum = 0; let evenMin = 0; let evenMax = 0;
    let oddMinPrevious = 0; let oddMaxPrevious = 0; let evenMinPrevious = 0; let evenMaxPrevious = 0;
    isOddNo = false; isEvenNo = false;

    if (numbers === 0) {isOddNo = true}
    if (numbers === 1) {isEvenNo = true}

    for (let i = 1; i <= numbers; i++) {
        currentNumber = Number(array[i]);

        if (i === 1) {
            oddSum += currentNumber; oddMaxPrevious = currentNumber; oddMinPrevious = currentNumber;
        } else if (i % 2 !==0) {
            oddSum += currentNumber;
            if (currentNumber >= oddMaxPrevious) {oddMaxPrevious = currentNumber;}
            if (currentNumber <= oddMinPrevious) {oddMinPrevious = currentNumber;}
        } else if (i === 2) {
            evenSum += currentNumber; evenMaxPrevious = currentNumber; evenMinPrevious = currentNumber;
        } else if (i % 2 === 0) {
            evenSum += currentNumber;
                if (currentNumber >= evenMaxPrevious) {evenMaxPrevious = currentNumber;}
                if (currentNumber <= evenMinPrevious) {evenMinPrevious = currentNumber;}
        }
    }
    
    oddSum = (oddSum).toFixed(2);
    oddMax = (oddMaxPrevious).toFixed(2);
    oddMin = (oddMinPrevious).toFixed(2);   
    evenSum = (evenSum).toFixed(2);
    evenMax = (evenMaxPrevious).toFixed(2);
    evenMin = (evenMinPrevious).toFixed(2);

    if (isOddNo) { oddMin = "No"; oddMax = "No"; evenMin = "No"; evenMax = "No";}
    if (isEvenNo) { evenMin = "No"; evenMax = "No";}

    console.log(`OddSum=${oddSum},`);
    console.log(`OddMin=${oddMin},`);
    console.log(`OddMax=${oddMax},`);
    console.log(`EvenSum=${evenSum},`);
    console.log(`EvenMin=${evenMin},`);
    console.log(`EvenMax=${evenMax}`);
}
    vladi(["6", "2", "3", "5", "4", "2", "1"]);
    // vladi(["5", "3", "-2", "8", "11", "-3"]);
    // vladi(["2", "1.5", "-2.5"]);
    // vladi(["4", "1.5", "1.75", "1.5", "1.75"]);
    // vladi(["1", "1"]);
    // vladi(["1", "5"]);
    // vladi(["0"]);
    // vladi(["3", "-1", "-2", "-3"]);